ddtrace/contrib/rq/__init__.py

```diff
@@ -0,0 +1,245 @@
+"""
+The RQ integration will trace your jobs.
+
+Usage
+~~~~~
+
+The rq integration is enabled automatically when using
+:ref:`ddtrace-run<ddtracerun>` or :ref:`patch_all()<patch_all>`.
+
+Or use :ref:`patch()<patch>` to manually enable the integration::
+
+    from ddtrace import patch
+    patch(rq=True)
+
+
+Worker Usage
+~~~~~~~~~~~~
+
+``ddtrace-run`` can be used to easily trace your workers::
+
+    DD_RQ_WORKER_SERVICE=myworker ddtrace-run rq worker
+
+
+See https://github.com/DataDog/trace-examples/tree/kyle-verhoog/rq/python/rq
+for a working example.
+
+
+Instance Configuration
+~~~~~~~~~~~~~~~~~~~~~~
+
+
+
+Global Configuration
+~~~~~~~~~~~~~~~~~~~~
+
+.. py:data:: ddtrace.config.rq['distributed_tracing_enabled']
+
+   If ``True`` the integration will connect the traces sent between the queuer
+   and the RQ worker.
+
+   Default: ``False``
+
+.. py:data:: ddtrace.config.rq['service']
+
+   The service name reported by default for RQ spans from the app.
+
+   This option can also be set with the ``DD_RQ_SERVICE`` or ``DD_SERVICE``
+   environment variables.
+
+   Default: ``rq``
+
+.. py:data:: ddtrace.config.rq_worker['service']
+
+   The service name reported by default for RQ spans from workers.
+
+   This option can also be set with the ``DD_RQ_WORKER_SERVICE`` environment
+   variable.
+
+   Default: ``rq-worker``
+"""
+from ddtrace import Pin
+from ddtrace import config
+
+from .. import trace_utils
+from ...ext import SpanTypes
+from ...propagation.http import HTTPPropagator
+from ...utils import get_argument_value
+
+
+__all__ = [
+    "patch",
+    "unpatch",
+]
+
+
+config._add(
+    "rq",
+    dict(
+        _default_service="rq",
+        distributed_tracing_enabled=False,
+    ),
+)
+
+config._add(
+    "rq_worker",
+    dict(
+        _default_service="rq-worker",
+    ),
+)
+
+
+@trace_utils.with_traced_module
+def traced_queue_enqueue_job(rq, pin, func, instance, args, kwargs):
+    job = get_argument_value(args, kwargs, 0, "f")
+
+    with pin.tracer.trace(
+        "rq.queue.enqueue_job",
+        service=trace_utils.int_service(pin, config.rq),
+        span_type=SpanTypes.WORKER,
+        resource=job.func_name,
```

Comment:
Looking at the RQ dashboard the monitoring users perform make reference to queue names and worker names.

I'm inclined to think that both these should show up on a span and not just as tags.

```suggestion
        resource="{} {}".format(instance.name, job.func_name),
```

Reply:
Yeah so rq does this by default for functional jobs but doesn't fully qualify the classes:

https://github.com/rq/rq/blob/c556106a38505ea7485daec9420c408f720d2961/rq/job.py#L105-L117

so I'll add support for when there's an `instance`